Pod/Classes/Optional Data Sources/VOKPagingFetchedResultsDataSource.m

```diff
@@ -97,57 +158,69 @@ - (void)observeValueForKeyPath:(NSString *)keyPath
-            newInsets.bottom += self.footerView.frame.size.height;
-            
-            [self triggerAction:self.downAction forAccessoryView:self.footerView withInsets:newInsets];
-        }
-        
-        CGFloat topOffset = scrollView.contentOffset.y - scrollView.contentInset.top;
-        if (topOffset < (-self.triggerDistance) && self.upAction) {
-            
-            UIEdgeInsets newInsets = self.orginalInsets;
-            newInsets.top += self.headerView.frame.size.height;
-            
-            [self triggerAction:self.upAction forAccessoryView:self.headerView withInsets:newInsets];
+        CGFloat distanceBelowBottomOfScreen = scrollView.contentSize.height - CGRectGetMaxY(scrollView.bounds);
+        CGFloat distanceAboveTopOfScreen = CGRectGetMinY(scrollView.bounds);
+
+        //Action for getting to bottom
+        if ([self contentHeightLargerThanFrameForScrollView:scrollView]
+            && distanceBelowBottomOfScreen <= self.triggerDistance
+            && self.downAction) {
+            CGFloat contentHeight = scrollView.contentSize.height;
+            CGFloat offsetHeight = contentHeight - CGRectGetHeight(self.tableView.frame) - CGRectGetHeight(self.footerView.frame);
+            CGPoint offset = CGPointMake(0, offsetHeight);
+            UIEdgeInsets insets = self.tableView.contentInset;
+            insets.bottom += DefaultAccessoryHeight;
+
+            [self triggerAction:self.downAction
+               forAccessoryView:self.footerView
+              withContentOffset:offset
+              returningToOffset:offset
+                temporaryInsets:insets];
+        } else if (distanceAboveTopOfScreen < (-self.triggerDistance) && self.upAction) {
+            //Action for pulling down from top.
+            CGPoint offset = CGPointMake(0, CGRectGetHeight(self.headerView.frame));
+            UIEdgeInsets insets = self.tableView.contentInset;
+            insets.top += DefaultAccessoryHeight;
+
+            [self triggerAction:self.upAction
+               forAccessoryView:self.headerView
+              withContentOffset:CGPointZero
+              returningToOffset:offset
+                temporaryInsets:insets];
         }
     }
 }
 
 - (void)scrollViewDidScroll:(UIScrollView *)scrollView
 {
     if (!self.isLoading) {
-        //Calculate scrollable height
-        CGFloat contentHeight = scrollView.contentSize.height;
-        CGFloat scrollableHeight = contentHeight - scrollView.bounds.size.height;
-        
-        CGFloat topOffset = scrollView.contentOffset.y + scrollView.contentInset.top;
-        
-        if (topOffset > scrollableHeight) {
-            CGFloat distanceOverscrolled = topOffset - scrollableHeight;
-            [self.footerView hasOverScrolled:(distanceOverscrolled/self.triggerDistance)];
+        CGFloat distanceBelowBottomOfScreen = scrollView.contentSize.height - CGRectGetMaxY(scrollView.bounds);
+        CGFloat distanceAboveTopOfScreen = CGRectGetMinY(scrollView.bounds);
```

Comment:
These two exist in at least two places—worth making them calculated properties in a category on `UIScrollView`?  (If so, maybe replace `contentHeightLargerThanFrameForScrollView:` with one, too.)